etcd3/client.py

```diff
@@ -155,17 +191,70 @@ def __init__(self, host='localhost', port=2379,
                 'must be specified.'
             )
 
-        self.kvstub = etcdrpc.KVStub(self.channel)
-        self.watcher = watch.Watcher(
-            etcdrpc.WatchStub(self.channel),
+        # The watcher is a special beast, as it spawns threads
+        self.watcher = self.get_watcher()
+        self.transactions = Transactions()
+
+    @property
```

Comment:
Not against this change but it really does not have anything to do with the intention of this patch, right?
It also makes new object each time which has a cost. I'd prefer to see this kind of change in a different PR.

Reply:
It does. You need to ensure the various stubs use the correct channel, which is the currently active one. If you don't do this, all these properties will contain a reference to the channel used at the time you created the object. The alternative approach would be to reinstantiate every single one of these objects on every reconnection, which is fine but it felt less elegant and error-prone to me.